flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/LaunchableMesosWorker.java

```
@@ -0,0 +1,205 @@
+
+import static org.apache.flink.mesos.Utils.variable;
+import static org.apache.flink.mesos.Utils.range;
+import static org.apache.flink.mesos.Utils.ranges;
+import static org.apache.flink.mesos.Utils.scalar;
+
+/**
+ * Specifies how to launch a Mesos worker.
+ */
+public class LaunchableMesosWorker implements LaunchableTask {
+
+	/**
+	 * The set of configuration keys to be dynamically configured with a port allocated from Mesos.
+	 */
+	private static String[] TM_PORT_KEYS = {
+		"taskmanager.rpc.port",
+		"taskmanager.data.port" };
+
+	private final MesosTaskManagerParameters params;
+	private final Protos.TaskInfo.Builder template;
+	private final Protos.TaskID taskID;
+	private final Request taskRequest;
+
+	/**
+	 * Construct a launchable Mesos worker.
+	 * @param params the TM parameters such as memory, cpu to acquire.
+	 * @param template a template for the TaskInfo to be constructed at launch time.
+	 * @param taskID the taskID for this worker.
+     */
+	public LaunchableMesosWorker(MesosTaskManagerParameters params, Protos.TaskInfo.Builder template, Protos.TaskID taskID) {
+		this.params = params;
+		this.template = template;
+		this.taskID = taskID;
+		this.taskRequest = new Request();
+	}
+
+	public Protos.TaskID taskID() {
+		return taskID;
+	}
+
+	@Override
+	public TaskRequest taskRequest() {
+		return taskRequest;
+	}
+
+	class Request implements TaskRequest {
+		private final AtomicReference<TaskRequest.AssignedResources> assignedResources = new AtomicReference<>();
+
+		@Override
+		public String getId() {
+			return taskID.getValue();
+		}
+
+		@Override
+		public String taskGroupName() {
+			return "";
+		}
+
+		@Override
+		public double getCPUs() {
+			return params.cpus();
+		}
+
+		@Override
+		public double getMemory() {
+			return params.containeredParameters().taskManagerTotalMemoryMB();
+		}
+
+		@Override
+		public double getNetworkMbps() {
+			return 0.0;
+		}
+
+		@Override
+		public double getDisk() {
+			return 0.0;
```

I would rather throw an exception here if the value is not in use.